src/vs/workbench/common/editor/filteredEditorGroupModel.ts

```diff
@@ -80,7 +72,7 @@ export class StickyEditorGroupModel extends FilteredEditorGroupModel {
 		if (options?.excludeSticky) {
 			return [];
 		}
-		return super.getEditors(order, options);
+		return this.model.getEditors(EditorsOrder.SEQUENTIAL).slice(0, this.model.stickyCount);
```

Comment:
Looks dangerous to me, I know we are just using the sequential order currently, but future changes might rely on other orders, how about just checking if the order that is passed actually is sequential, then we do the efficient solution and otherwise we `filter()` the editors